### PR TITLE
fix: Popups render in the target's window

### LINKED ISSUE #2831: Popup content rendered in a portalled external window shows popup contents in parent window
## 🐛 Bug Report

If `window.open` is used to open an `about:blank` page and React portals are used to render content into that window, and Popup in the external window is rendered in the parent window. The popper will render in the correct location that it should render in the child external window. In other words, the popper will render at the correct x,y coordinates, but in the wrong window.

## To Reproduce

Render a Popup using `window.open('', '', 'popup=true')` and portal content into that window's reference and render a `Tooltip` there.

## Expected Behavior

The Tooltip renders in the window the target is located in.

## Actual Results

The Tooltip renders in the parent window at the exact coordinates it should render in the external window.

## Link to repl or repo (highly encouraged)

Unfortunately, this type of behavior doesn't work in a code sandbox.

### Error Output

Tooltip of a child window rendering in the parent window

### PR DESCRIPTION
**Pull request by NicholasBoll**
## Summary

Popups now render in the same window as their target element.

Fixes #2831 

## Release Category
Components
---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [x] Label `ready for review` has been added to PR

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)

## Where Should the Reviewer Start?

Unfortunately, opening contents in a new window doesn't work within an `iframe` of a code sandbox correctly, so there cannot be an example code sandbox. I created a new example to showcase this behavior: https://5d854c26ba934e0020f5e98a-fgaxeeknap.chromatic.com/iframe.html?args=&id=components-popups-popup--external-window&viewMode=story

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation

## Screenshots or GIFs (if applicable)

[PopupExternalWindow.webm](https://github.com/user-attachments/assets/c40897e0-3cb5-4d89-853c-3fa6677e2698)

### CODE REVIEW

File: modules/react/popup/stories/examples/ExternalWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      setPortalElement(element);
+    } else {
+      onWindowClose();
+    }
+
+    const closeWindow = event => {
+      onWindowClose();
+    };
+
+    // PopupStack.pushStackContext(newWindow.document.body);
```
Comment (mannycarrera4) [inline modules/react/popup/stories/examples/ExternalWindow.tsx:109]:
```suggestion
```

File: modules/react/popup/stories/examples/ExternalWindow.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+    window.addEventListener('unload', closeWindow);
+    newWindow?.addEventListener('unload', closeWindow);
+
+    return () => {
+      // PopupStack.popStackContext(newWindow.document.body);
```
Comment (mannycarrera4) [inline modules/react/popup/stories/examples/ExternalWindow.tsx:115]: do you want to remove this?

Comment (mannycarrera4) [inline modules/react/popup/stories/examples/ExternalWindow.tsx:115]:
```suggestion
```

**Review by mannycarrera4 — APPROVED** [review]

File: modules/react/popup/stories/examples/ExternalWindow.tsx (hunk 2 above)
Comment (NicholasBoll) [reply modules/react/popup/stories/examples/ExternalWindow.tsx]: Yes. Thank you

**Pull request merged**